### PR TITLE
ref(invite): add people form

### PR DESCRIPTION
**Pull request by gabiborlea**
Screenshot 2023-04-13 at 14 13 51
Screenshot 2023-04-13 at 14 13 44
Screenshot 2023-04-13 at 14 13 22

### CODE REVIEW

File: react/features/base/react/components/web/InlineDialogFailure.tsx [hunk 1]
```diff
@@ -1,13 +1,33 @@
-import React, { Component } from 'react';
-import { WithTranslation } from 'react-i18next';
+/* @flow */
```
Comment (saghul) [inline react/features/base/react/components/web/InlineDialogFailure.tsx:1]: :fire:

File: react/features/base/react/components/web/MultiSelectAutocomplete.tsx [hunk 2]
```diff
@@ -274,21 +274,22 @@ class MultiSelectAutocomplete extends Component<Props, State> {
         );
 
         return (
-            <AKInlineDialog
+
+            // @ts-ignore
```
Comment (saghul) [inline react/features/base/react/components/web/MultiSelectAutocomplete.tsx:278]: why do you need to ts-ignore here?

File: react/features/base/ui/components/web/ContextMenu.tsx [hunk 3]
```diff
@@ -119,6 +119,8 @@ interface IProps {
      * Tab index for the menu.
      */
     tabIndex?: number;
+
+    useEntityWidth?: boolean;
```
Comment (saghul) [inline react/features/base/ui/components/web/ContextMenu.tsx:123]: Pl document this. Also, wouldn't something like `containerWidth` be more descriptive?

**Review by saghul — COMMENTED** [review]
Left a couple of comments, but I think @robertpin should also take a look here.

File: react/features/base/react/components/web/InlineDialogFailure.tsx [hunk 4]
```diff
@@ -1,13 +1,33 @@
...
+            }
+        }
+    };
+});
+
 /**
  * The type of the React {@code Component} props of {@link InlineDialogFailure}.
  */
-interface IProps extends WithTranslation {
+type Props = {
```
Comment (robertpin) [inline react/features/base/react/components/web/InlineDialogFailure.tsx:30]: Please leave it as interface. We generally try to use interfaces everywhere and types only where we can't use interfaces

File: react/features/base/react/components/web/InlineDialogFailure.tsx [hunk 5]
```diff
@@ -1,13 +1,33 @@
...
+    return {
+        dialog: {
+            backgroundColor: theme.palette.ui01,
+            border: `1px solid ${theme.palette.ui04}`,
+            borderRadius: `${Number(theme.shape.borderRadius)}px`,
+            boxShadow: '0px 1px 2px rgba(41, 41, 41, 0.25)',
+            color: theme.palette.text01,
+            ...withPixelLineHeight(theme.typography.bodyShortRegular),
+            padding: `${theme.spacing(3)} 10`,
+            '& button': {
```
Comment (robertpin) [inline react/features/base/react/components/web/InlineDialogFailure.tsx:20]: Please make this a separate class instead of using the tag selector

File: react/features/base/react/components/web/MultiSelectAutocomplete.tsx [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
-import { MultiSelectStateless } from '@atlaskit/multi-select';
 import _debounce from 'lodash/debounce';
 import React, { Component } from 'react';
 
+import Popover from '../../../popover/components/Popover.web';
+import { MultiSelectItem } from '../../../ui/components/types';
+import MultiSelect from '../../../ui/components/web/MultiSelect';
 import logger from '../../logger';
 
+// @ts-ignore
```
Comment (robertpin) [inline react/features/base/react/components/web/MultiSelectAutocomplete.tsx:9]: Why is this necessary?

File: react/features/base/react/components/web/MultiSelectAutocomplete.tsx [hunk 7]
```diff
@@ -85,34 +85,34 @@ type Props = {
 type State = {
```
Comment (robertpin) [inline react/features/base/react/components/web/MultiSelectAutocomplete.tsx:85]: Please convert to interface

File: react/features/invite/components/add-people-dialog/AbstractAddPeopleDialog.tsx [hunk 8]
```diff
@@ -3,29 +3,30 @@ import { Component } from 'react';
 import { createInviteDialogEvent } from '../../../analytics/AnalyticsEvents';
 import { sendAnalytics } from '../../../analytics/functions';
 import { IReduxState } from '../../../app/types';
-import { showNotification } from '../../../notifications/actions';
+import { showErrorNotification, showNotification } from '../../../notifications/actions';
 import { NOTIFICATION_TIMEOUT_TYPE } from '../../../notifications/constants';
-import { INotificationProps } from '../../../notifications/types';
-import { invite } from '../../actions';
+import { invite } from '../../actions.web';
```
Comment (robertpin) [inline react/features/invite/components/add-people-dialog/AbstractAddPeopleDialog.tsx:8]: Don't import from `.web` on Abstract components

File: react/features/invite/components/add-people-dialog/AbstractAddPeopleDialog.tsx [hunk 9]
```diff
@@ -3,29 +3,30 @@ import { Component } from 'react';
...
     isAddPeopleEnabled,
     isDialOutEnabled,
     isSipInviteEnabled
 } from '../../functions';
 import logger from '../../logger';
-import { IInvitee } from '../../types';
+import { IInvitee, InviteSelectItem } from '../../types';
 
-export interface IProps {
+export type Props = {
```
Comment (robertpin) [inline react/features/invite/components/add-people-dialog/AbstractAddPeopleDialog.tsx:20]: Interface please

File: react/features/invite/components/add-people-dialog/AbstractAddPeopleDialog.tsx [hunk 10]
```diff
@@ -48,33 +49,33 @@ export interface IProps {
...
 
     /**
      * The Redux dispatch function.
      */
     dispatch: Function;
-}
+};
 
-export interface IState {
+export type State = {
```
Comment (robertpin) [inline react/features/invite/components/add-people-dialog/AbstractAddPeopleDialog.tsx:78]: Interface please

File: react/features/invite/components/add-people-dialog/AbstractAddPeopleDialog.tsx [hunk 11]
```diff
@@ -106,13 +106,13 @@ export default class AbstractAddPeopleDialog<P extends IProps, S extends IState>
     constructor(props: P) {
         super(props);
 
-        this._query = this._query.bind(this);
+        this._query = this._query?.bind(this);
```
Comment (robertpin) [inline react/features/invite/components/add-people-dialog/AbstractAddPeopleDialog.tsx:109]: Why use optional chaining here?

File: react/features/invite/components/add-people-dialog/AbstractAddPeopleDialog.tsx [hunk 12]
```diff
@@ -176,19 +176,18 @@ export default class AbstractAddPeopleDialog<P extends IProps, S extends IState>
...
-
-                    this.setState({
-                        addToCallError: true
-                    });
+                    dispatch(showErrorNotification({
+                        titleKey: 'addPeople.failedToAdd'
+                    }, NOTIFICATION_TIMEOUT_TYPE.MEDIUM));
                 } else if (!_callFlowsEnabled) {
                     const invitedCount = invitees.length;
-                    let notificationProps: INotificationProps | undefined;
```
Comment (robertpin) [inline react/features/invite/components/add-people-dialog/AbstractAddPeopleDialog.tsx:185]: Why was this removed? Was it wrong or it's unnecessary?

File: react/features/invite/types.ts [hunk 13]
```diff
@@ -1,6 +1,20 @@
...
+    name?: string;
     number: string;
+    originalEntry?: string;
+    phone?: string;
+    showCountryCodeReminder?: boolean;
     type: string;
+    user_id?: string;
 }
+
+export type InviteSelectItem = MultiSelectItem & {
```
Comment (robertpin) [inline react/features/invite/types.ts:16]: Please use interfaces with `extends`

File: react/features/invite/components/add-people-dialog/web/InviteContactsForm.tsx [hunk 14]
```diff
@@ -1,63 +1,92 @@
...
     type Props as AbstractProps,
     type State,
     _mapStateToProps as _abstractMapStateToProps
 } from '../AbstractAddPeopleDialog';
 
-declare var interfaceConfig: Object;
+const styles = () => {
+    return {
+        formWrap: {
+            marginTop: '8px'
```
Comment (robertpin) [inline react/features/invite/components/add-people-dialog/web/InviteContactsForm.tsx:26]: Please use `theme.spacing`

File: react/features/invite/components/add-people-dialog/web/InviteContactsForm.tsx [hunk 15]
```diff
@@ -1,63 +1,92 @@
...
+        },
+        inviteButtons: {
+            display: 'flex',
+            justifyContent: 'end',
+            marginTop: '8px',
```
Comment (robertpin) [inline react/features/invite/components/add-people-dialog/web/InviteContactsForm.tsx:31]: Please use `theme.spacing`

File: react/features/invite/components/add-people-dialog/web/InviteContactsForm.tsx [hunk 16]
```diff
@@ -1,63 +1,92 @@
...
+            '&>button': {
```
Comment (robertpin) [inline react/features/invite/components/add-people-dialog/web/InviteContactsForm.tsx:32]: Could we use some class instead please? Tag selectors are not the best for performance

File: react/features/invite/components/add-people-dialog/web/InviteContactsForm.tsx [hunk 17]
```diff
@@ -1,63 +1,92 @@
...
+                marginLeft: '8px'
+            }
+        }
+    };
+};
+
 
 type Props = AbstractProps & {
```
Comment (robertpin) [inline react/features/invite/components/add-people-dialog/web/InviteContactsForm.tsx:40]: Please convert to interface

File: react/features/base/ui/components/web/MultiSelect.tsx [hunk 18]
```diff
@@ -0,0 +1,174 @@
...
+import Popover from '../../../popover/components/Popover.web';
+import { MultiSelectItem } from '../types';
+
+import ClickableIcon from './ClickableIcon';
+import ContextMenu from './ContextMenu';
+import ContextMenuItem from './ContextMenuItem';
+import ContextMenuItemGroup from './ContextMenuItemGroup';
+import Input from './Input';
+
+type Props = {
```
Comment (robertpin) [inline react/features/base/ui/components/web/MultiSelect.tsx:14]: Please convert to interface

File: react/features/base/ui/components/web/MultiSelect.tsx [hunk 19]
```diff
@@ -0,0 +1,174 @@
...
+    onRemoved: (item: any) => void;
+    onSelected: (item: any) => void;
+    placeholder?: string;
+    selectedItems?: MultiSelectItem[];
+};
+
+const useStyles = makeStyles()(theme => {
+    return {
+        contextMenu: {
+            position: 'relative' as const,
```
Comment (robertpin) [inline react/features/base/ui/components/web/MultiSelect.tsx:32]: `as const` should only be necessary when using `withStyles`. For `makeStyles` it shouldn't be necessary

File: react/features/base/ui/components/web/MultiSelect.tsx [hunk 20]
```diff
@@ -0,0 +1,174 @@
...
+            marginTop: '8px',
```
Comment (robertpin) [inline react/features/base/ui/components/web/MultiSelect.tsx:33]: Please use `theme.spacing`

File: react/features/base/ui/components/web/MultiSelect.tsx [hunk 21]
```diff
@@ -0,0 +1,174 @@
...
+                backgroundColor: theme.palette.ui01
+            },
+            cursor: 'auto',
+            paddingTop: '0px',
+            paddingBottom: '0px',
+            paddingLeft: '16px',
+            paddingRight: '6px'
+        },
+        contentListItem: {
+            inlineSize: 'calc(100% - 38px)',
```
Comment (robertpin) [inline react/features/base/ui/components/web/MultiSelect.tsx:59]: Why 38px? Where does it come from?

File: react/features/base/ui/components/web/MultiSelect.tsx [hunk 22]
```diff
@@ -0,0 +1,174 @@
...
+            overflowWrap: 'break-word'
+        },
+        contentSelectedItem: {
+            inlineSize: 'calc(100% - 82px)',
```
Comment (robertpin) [inline react/features/base/ui/components/web/MultiSelect.tsx:63]: Same question as above

File: react/features/base/ui/components/web/MultiSelect.tsx [hunk 23]
```diff
@@ -0,0 +1,174 @@
...
+                                    accessibilityLabel = { 'multi-select-unselect' }
+                                    icon = { IconCloseLarge }
+                                    id = 'modal-header-close-button'
+                                    onClick = { removeItem(item) } />
+
+                            </ContextMenuItem>))
+                    }
+                </ContextMenuItemGroup>
+
+
```
Comment (robertpin) [inline react/features/base/ui/components/web/MultiSelect.tsx:167]: Please remove these 2 empty lines

File: react/features/base/ui/components/web/MultiSelect.tsx [hunk 24]
```diff
@@ -0,0 +1,174 @@
...
+                                <div className = { classes.contentListItem }>
+                                    {noMatchesText}
+                                </div>
+                            </ContextMenuItem>
+                        )
+                }
+            </ContextMenuItemGroup>
+        </ContextMenu>
+    )
+    , [ items ]);
```
Comment (robertpin) [inline react/features/base/ui/components/web/MultiSelect.tsx:120]: Please put this on the line above. It looks weird like this :D

**Review by robertpin — CHANGES_REQUESTED** [review]

File: react/features/base/ui/components/web/MultiSelect.tsx (hunk 22 above)
Comment (gabiborlea) [reply react/features/base/ui/components/web/MultiSelect.tsx:63]: the size of the clear icon and the spacing

Comment (gabiborlea) [reply react/features/base/ui/components/web/MultiSelect.tsx:63]: the size of the icon + the clear icon and the spacing

**Review by robertpin — APPROVED** [review]
